src/main/java/gregtech/api/GregTechAPI.java

```diff
@@ -51,6 +51,8 @@ public class GregTechAPI {
     public static IAdvancementManager advancementManager;
     /** Will be available at the Pre-Initialization stage */
     public static ISoundManager soundManager;
+    /** Will be available at the Pre-Initialization stage */
+    public static boolean highTier;
```

Comment:
This should definitely not be kept as "public", especially given that the goal of this PR is to make sure that high-tier content cannot be forced off in any circumstance once turned on, which should certainly include other mods.